components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/confirmation/ResendConfirmationManager.java

```diff
@@ -418,8 +418,8 @@ private NotificationResponseBean resendAccountRecoveryNotification(User user, St
             }
         }
         String secretKey = generateSecretKey(preferredChannel);
-        UserRecoveryData recoveryDataDO = new UserRecoveryData(user, secretKey, RecoveryScenarios
-                .SELF_SIGN_UP, RecoverySteps.CONFIRM_SIGN_UP);
+        UserRecoveryData recoveryDataDO = new UserRecoveryData(user, secretKey, RecoveryScenarios.valueOf
```

Comment:
@somindatommy pleaser review this change

Reply:
This method requires enums, right?
Can we user `RecoveryScenarios getRecoveryScenario(String scenarioName)` and `RecoverySteps getRecoveryStep(String stepName)`?